src/JsCodeExtractor.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace WP_CLI\I18n;
+
+use Gettext\Extractors\JsCode;
+use Gettext\Translations;
+use Peast\Syntax\Exception as PeastException;
+use RecursiveCallbackFilterIterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use WP_CLI;
+
+class JsCodeExtractor extends JsCode {
```

Comment:
Class should be `final` and protected methods/properties should be private (unless they were already protected in the superclass).

Reply:
Generally, we don't want to add interfaces when this is not necessary. An interface is always a maintenance burden. As people can easily extend `JsCode` themselves, keeping `JsCodeExtractor` extensible is not needed.